matrix_sdk_base/src/client.rs

```diff
@@ -819,13 +828,13 @@ impl BaseClient {
     pub async fn receive_left_timeline_event(
         &self,
         room_id: &RoomId,
-        event: &EventJson<RoomEvent>,
+        event: &EventJson<AnyRoomEventStub>,
     ) -> Result<bool> {
         match event.deserialize() {
             Ok(e) => {
                 let room_lock = self.get_or_create_left_room(room_id).await?;
                 let mut room = room_lock.write().await;
-                Ok(room.receive_timeline_event(&e))
+                Ok(room.receive_timeline_event(&e, room_id))
```

Comment:
Why is necessary to pass the room id to the room, surely the room knows its own id already?